core/src/reporter/state.ts

```diff
@@ -294,38 +301,53 @@ export class State {
         throw new OraklError(OraklErrorCode.WalletNotActive, msg)
       }
 
-      let remoteNonce: number
-      try {
-        if (this.delegatedFee) {
-          const caverWallet = wallet as CaverWallet
-          remoteNonce = Number(
-            await caverWallet.caver.rpc.klay.getTransactionCount(caverWallet.address)
-          )
-        } else {
-          remoteNonce = await (wallet as NonceManager).getTransactionCount()
+      // Assumption: the only source of error can be json-rpc call
+      // Solution: keep polling/retrying until json-rpc becomes responsive
+      // If successful, the "return" statement will break the infinite loop
+      let retryCount = 0
+      while (true) {
+        try {
+          let remoteNonce: number
+          if (this.delegatedFee) {
+            const caverWallet = wallet as CaverWallet
+            remoteNonce = Number(
+              await caverWallet.caver.rpc.klay.getTransactionCount(caverWallet.address)
+            )
+          } else {
+            remoteNonce = await (wallet as NonceManager).getTransactionCount()
+          }
+
+          const localNonce = this.nonces[oracleAddress]
+
+          let nonce: number
+          if (!localNonce || remoteNonce > localNonce) {
+            this.logger.warn(
+              { name: 'getAndIncrementNonce', file: FILE_NAME },
+              `Nonce value discrepancy. Remote: ${remoteNonce}, Local: ${localNonce}. Updating local nonce to remote nonce.`
+            )
+            nonce = remoteNonce
+          } else {
+            nonce = localNonce
+          }
+
+          this.nonces[oracleAddress] = nonce + 1
+
+          return nonce
+        } catch (error) {
+          // Slack the error message every half an hour
+          if (retryCount % NONCE_MANAGER_SLACK_FREQUENCY_RETRIES === 0) {
+            this.logger.error(
+              { name: 'getAndIncrementNonce', file: FILE_NAME },
+              `Error while fetching nonce for oracle ${oracleAddress}. Retrying...`
+            )
+            await sendToSlack(error)
+          }
+          retryCount += 1
+
+          // Sleep before retrying
+          await sleep(NONCE_MANAGER_POLLING_INTERVAL)
```

Comment:
infinite loop is an intentional feature. This will prevent bullmq from marking in-progress jobs as failed. Once json rpc becomes responsive again, it'll fulfill those requests.